frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-builder/update-utils.ts

```diff
@@ -431,6 +512,25 @@ const fixInvalidListTask: UpdateOperationAction<UpdateOperationFixInvalidTaskLis
     ],
     listTasks,
     errors: getErrors(newPipelineTask, resource),
+    finallyTasks,
+    finallyListTasks,
+  };
+};
+
+const addFinallyListTask: UpdateOperationAction<UpdateOperationFixInvalidTaskListData> = (
+  tasks,
+  listTasks,
+  data,
+  finallyTasks,
+  finallyListTasks,
+) => {
+  const newTaskName = `finally-list-${getRandomChars(6)}`;
+
+  return {
+    finallyTasks: [...finallyTasks],
+    finallyListTasks: [...finallyListTasks, { name: newTaskName }],
+    tasks: [...tasks],
```

Comment:
Not sure you need to re-wrap the items... omit spread? (and if you do one of my previous changes, omit entirely)